src/axom/slam/policies/IndirectionPolicies.hpp

```diff
@@ -105,10 +105,14 @@ struct IndexedIndirection : public BasePolicy
     __trap();
   #elif defined(__HIP_DEVICE_COMPILE__)
     abort();
+  #else
+    // Always return a value.
+    return buf[pos];
   #endif
-#endif
+#else
     // Always return a value.
     return buf[pos];
+#endif
```

Comment:
Is this change meant to be more explicit? 

My read is that the previous behavior always called `return buf[pos]` (regardless of whether `AXOM_DEVICE_CODE` is defined) and the updated behavior does the same.

Reply:
I suppose I could have removed the comment " Always return a value". The problem for the CUDA/HIP branches of these functions was that an STLVectorIndirection was passed into the IndexedIndirection and the operator[] it ends up using is not a host-device function and never could be, resulting in a lot of compilation warnings on rzansel.